server_windows.go

```diff
@@ -37,3 +43,86 @@ func (s *Server) toLocalPath(p string) string {
 
 	return lp
 }
+
+func bitsToDrives(bitmap uint32) []string {
+	var drive rune = 'a'
+	var drives []string
+
+	for bitmap != 0 {
+		if bitmap&1 == 1 {
+			drives = append(drives, string(drive)+":")
+		}
+		drive++
+		bitmap >>= 1
+	}
+
+	return drives
+}
+
+func getDrives() ([]string, error) {
+	mask, err := windows.GetLogicalDrives()
+	if err != nil {
+		return nil, fmt.Errorf("GetLogicalDrives: %w", err)
+	}
+	return bitsToDrives(mask), nil
+}
+
+type driveInfo struct {
+	fs.FileInfo
+	name string
+}
+
+func (i *driveInfo) Name() string {
+	return i.name // since the Name() returned from a os.Stat("C:\\") is "\\"
+}
+
+type winRoot struct {
+	dummyFile
+	drives []string
+}
+
+func newWinRoot() (*winRoot, error) {
+	drives, err := getDrives()
+	if err != nil {
+		return nil, err
+	}
+	return &winRoot{
+		drives: drives,
+	}, nil
+}
+
+func (f *winRoot) Readdir(n int) ([]os.FileInfo, error) {
+	drives := f.drives
+	if n > 0 {
+		if len(drives) > n {
+			drives = drives[:n]
+		}
+		f.drives = f.drives[len(drives):]
+		if len(drives) == 0 {
+			return nil, io.EOF
+		}
```

Comment:
This code needs to be outside of the `if n > 0` block.

Otherwise, a `Readdir(-1)` would just continuously return all the drives forever.